test/cases/api.sh

```diff
@@ -953,4 +953,35 @@ curl \
         --header "x-rh-identity: $INVALIDAUTHSTRING" \
         http://localhost:443/api/composer/v1/version)" = "404" ]
 
+#
+# Make sure that requesting a non existing paquet returns a 400 error
+#
+[ "$(curl \
+    --silent \
+    --output /dev/null \
+    --write-out '%{http_code}' \
+    --header "x-rh-identity: $VALIDAUTHSTRING" \
+    -H "Content-Type: application/json" \
+    -d '{ "distribution": "centos-8", "image_requests": [ { "architecture": "x86_64", "image_type": "ami", "repositories": [ { "baseurl": "http://mirror.centos.org/centos/8-stream/BaseOS/x86_64/os/", "rhsm": false }, { "baseurl": "http://mirror.centos.org/centos/8-stream/AppStream/x86_64/os/", "rhsm": false }, { "baseurl": "http://mirror.centos.org/centos/8-stream/extras/x86_64/os/", "rhsm": false } ], "upload_request": { "type": "aws.s3", "options": { "region": "somewhere", "s3": { "access_key_id": "thingy", "secret_access_key": "thing", "bucket": "thingything" } } } } ], "customizations": { "packages": [ "jesuisunpaquetquinexistepas_idonotexist" ] } }' \
+    http://localhost:443/api/composer/v1/compose)" = "400" ]
+
+#
+# Make sure that a request that makes the dnf-json crash returns a 500 error
+#
+sudo cp -f /usr/libexec/osbuild-composer/dnf-json /usr/libexec/osbuild-composer/dnf-json.bak
+sudo cat << EOF | sudo tee /usr/libexec/osbuild-composer/dnf-json
+#!/usr/bin/python3
+raise Exception
+EOF
+[ "$(curl \
+    --silent \
+    --output /dev/null \
+    --write-out '%{http_code}' \
+    --header "x-rh-identity: $VALIDAUTHSTRING" \
+    -H "Content-Type: application/json" \
+    -d '{"distribution":"centos-8","image_requests":[{"architecture":"x86_64","image_type":"ami"}]}' \
```

Comment:
One last nitpick: can this be a valid request? That way we're sure the 500 is because of the dnf crash :)

Reply:
Definitely !